packages/react-charting/src/components/GaugeChart/GaugeChart.base.tsx

```diff
@@ -0,0 +1,627 @@
+        : `${this._sweepFraction[0]}/${this._sweepFraction[1]}`);
+    let total = this._minValue - 1;
+    const hoverYValues: IYValue[] = this._segments.map(segment => {
+      const yValue: IYValue = {
+        legend: segment.legend,
+        y: `${total + 1} - ${total + segment.size}`,
+        color: segment.color,
+      };
+      total += segment.size;
+      return yValue;
+    });
+    this.setState({
+      focusedElement,
+      calloutTarget: focusEvent.target,
+      isCalloutVisible: true,
+      hoverXValue,
+      hoverYValues,
+    });
+  };
+
+  private _handleBlur = () => {
+    this.setState({
+      focusedElement: '',
+      calloutTarget: null,
+      isCalloutVisible: false,
+      hoverXValue: '',
+      hoverYValues: [],
+    });
+  };
+
+  private _handleCalloutDismiss = () => {
+    this.setState({ isCalloutVisible: false });
+  };
+
+  private _handleMouseOver = (mouseEvent: React.MouseEvent<SVGElement>, hoveredElement: string) => {
+    if (this._calloutAnchor === hoveredElement) {
+      return;
+    }
+
+    this._calloutAnchor = hoveredElement;
+    const hoverXValue: string =
+      'Current value is ' +
+      (this.props.chartValueFormat === GaugeValueFormat.Fraction
+        ? `${((this._sweepFraction[0] / this._sweepFraction[1]) * 100).toFixed()}%`
+        : `${this._sweepFraction[0]}/${this._sweepFraction[1]}`);
+    let total = this._minValue - 1;
+    const hoverYValues: IYValue[] = this._segments.map(segment => {
+      const yValue: IYValue = {
+        legend: segment.legend,
+        y: `${total + 1} - ${total + segment.size}`,
+        color: segment.color,
+      };
+      total += segment.size;
+      return yValue;
+    });
+    this.setState({
+      calloutTarget: mouseEvent.target,
+      isCalloutVisible: true,
+      hoverXValue,
+      hoverYValues,
+    });
+  };
+
+  private _handleMouseOut = () => {
+    this._calloutAnchor = '';
+    this.setState({ calloutTarget: null, isCalloutVisible: false, hoverXValue: '', hoverYValues: [] });
+  };
+
+  private _wrapContent = (content: string, id: string, maxWidth: number) => {
+    const text = d3Select<SVGTextElement, {}>(`#${id}`);
+    text.text(content);
+    if (!text.node()) {
+      return false;
+    }
+
+    let textOverflow = false;
+    let textLength = text.node()!.getComputedTextLength();
+    while (textLength > maxWidth && content.length > 0) {
+      content = content.slice(0, -1);
+      text.text(content + '...');
+      textOverflow = true;
+      textLength = text.node()!.getComputedTextLength();
+    }
+    return textOverflow;
+  };
+
+  // TO DO: Write a common functional component for Multi value callout and divide sub count method
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private _multiValueCallout = (calloutProps: any) => {
+    const yValueHoverSubCountsExists: boolean = this._yValueHoverSubCountsExists(calloutProps.YValueHover);
```

Comment:
we can add an example for this case.

Reply:
The bottom 3 functions are copied from CartesianChart component to render data for multiple legends in the hover card